core/scene.c

```diff
@@ -420,47 +432,35 @@ void scene_add_map(Scene *sc, Shape *map) {
     transform_set_parent(sc->map, sc->root, true);
 
 #if DEBUG_SCENE_EXTRALOG
-    cclog_debug("🏞 map added to the scene");
+    cclog_debug("🏞 map %p (id: %d) added to scene %p", sc->map, transform_get_id(sc->map), sc);
 #endif
 }
 
 Transform *scene_get_map(Scene *sc) {
     return sc->map;
 }
 
-void scene_remove_map(Scene *sc) {
-    vx_assert(sc != NULL);
-
-    if (sc->map != NULL) {
-        transform_remove_parent(sc->map, true);
-        sc->map = NULL;
+bool scene_remove_transform(Scene *sc, Transform *t, const bool keepWorld) {
+    if (sc == NULL || t == NULL) {
+        return false;
     }
 
+    if (transform_remove_parent(t, keepWorld)) {
+        _scene_register_removed_transform(sc, t);
 #if DEBUG_SCENE_EXTRALOG
-    cclog_debug("🏞 map removed from the scene");
-#endif
-}
-
-void scene_remove_transform(Scene *sc, Transform *t) {
-    vx_assert(sc != NULL);
-    if (t == NULL)
-        return;
-
-    scene_register_removed_transform(sc, t);
-    transform_remove_parent(t, true);
-
-#if DEBUG_SCENE_EXTRALOG
-    cclog_debug("🏞 transform removed from the scene");
+        cclog_debug("🏞 transform %p (id: %d) removed from scene %p", t, transform_get_id(t), sc);
 #endif
+        return true;
+    }
+    return false;
 }
 
-void scene_register_removed_transform(Scene *sc, Transform *t) {
+void scene_register_managed_transform(Scene *sc, Transform *t) {
     if (sc == NULL || t == NULL) {
         return;
     }
 
-    transform_retain(t);
-    fifo_list_push(sc->removed, t);
+    transform_set_managed_ptr(t, sc->game);
```

Comment:
Couldn't we use the scene pointer instead?
Core can't be built as standalone if we refer CGame.